Windows/CommonDialog/MainWindow.xaml.cs

```diff
@@ -0,0 +1,166 @@
+                Filter = "Text files (*.txt)|*.txt"
+            };
+
+            // Setting this flag to true will prompt the user if the file is going to be created.
+            saveFileDialog.CreatePrompt = true;
+
+            // Setting this flag to true will prompt the user if the file is going to be overwritten.
+            saveFileDialog.OverwritePrompt = true;
+
+            // Setting this flag to true will restore the Application.CurrentDirectory.
+            //saveFileDialog.RestoreDirectory = true;
+
+            // We can attach an event handler to the FileOk event of the dialog box.
+            saveFileDialog.FileOk += SaveFileDialog_FileOk;
+
+            string text = "";
+            if (saveFileDialog.ShowDialog() == true)
+            {
+                File.WriteAllText(saveFileDialog.FileName, "Saving File Successful");
+
+                text = $" Selected File Name : {saveFileDialog.FileName}\nSaved the file with a small text \" Saving File Successful \"";
+            }
+            else
+            {
+                text = "No file selected. \nEither cancel button was clicked or dialog was closed";
+            }
+
+            // Setting the data context of this application. Not related to dialog functionality.
+            DataModel data = DataModel.GetDialogRelatedInfo(saveFileDialog);
+            data.ResultBody = text;
+            DataContext = data;
+        }
+
+        private void SaveFileDialog_FileOk(object? sender, CancelEventArgs e)
+        {
+            // This can be used to perform addtional validation on the selected file.
+            if(sender is SaveFileDialog sfd)
+            {
+                string selectedFile = sfd.FileName;
+                if (selectedFile.Contains(" "))
+                {
+                    MessageBox.Show("File name cannot contain spaces. Please rename the file and try again.");
+                    e.Cancel = true;
+                }
+            }
+        }
+
+        private void Button_OpenFolderDialog(object sender, RoutedEventArgs e)
+        {
+            //OpenFolderDialog openFolderDialog = new OpenFolderDialog()
+            //{
+            //    Title = "Select folder to open ...",
+            //    InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles),
+            //};
+
+            //openFolderDialog.Multiselect = true;
+
+            //string text = "";
+            //if (openFolderDialog.ShowDialog() == true)
+            //{
+            //    text = "List of all subdirectories in the selected folders : \n\n";
+            //    foreach (var folder in openFolderDialog.FolderNames)
+            //    {
+            //        DirectoryInfo d = new DirectoryInfo(folder);
+            //        text += folder + "\n---\n";
+
+            //        d.GetDirectories().ToList().ForEach(x => text += x.FullName + "\n");
```

Comment:
This is not nice. Also possibly unnecessary, as the FolderNames already contain full names, don't they?

Reply:
I am listing subdirectories of the selected directory here and not the selected directories themselves,